Error `ECONNRESET` for validating some of the external links

I have the following snippet in my markdown file:

```markdown
Navigate to [MySQL distribution](https://dev.mysql.com/downloads/mysql/) to install MySQL `5.7`.
```

This rule keep failing with following error message:

```txt
  60:34  error  https://dev.mysql.com/downloads/mysql/ is dead. (request to https://dev.mysql.com/downloads/mysql/ failed, reason: connect ECONNRESET 137.254.60.11:443)  no-dead-link
```

The link, https://dev.mysql.com/downloads/mysql/, is valid but not sure why the rule is unable to verify this. When I try `curl -v https://dev.mysql.com/downloads/mysql/` from my terminal, I get the following headers + the site's html responses:

```txt
*   Trying 137.254.60.11...
* TCP_NODELAY set
* Connected to dev.mysql.com (137.254.60.11) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=US; ST=California; L=Redwood City; O=Oracle Corporation; OU=Production Engineering and Operation; CN=www.mysql.com
*  start date: Jan 25 00:00:00 2019 GMT
*  expire date: Mar 25 12:00:00 2020 GMT
*  subjectAltName: host "dev.mysql.com" matched cert's "dev.mysql.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert SHA2 Secure Server CA
*  SSL certificate verify ok.
> GET /downloads/mysql/ HTTP/1.1
> Host: dev.mysql.com
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Fri, 21 Jun 2019 14:17:44 GMT
< Server: Apache
< X-Frame-Options: SAMEORIGIN
< Strict-Transport-Security: max-age=15768000
< Set-Cookie: MySQL_S=4rb6luvlhcf1lteuqv1es0l54c9n0c9n; path=/; domain=mysql.com; HttpOnly
< Expires: Thu, 19 Nov 1981 08:52:00 GMT
< Cache-Control: no-store, no-cache, must-revalidate
< Pragma: no-cache
< Cache-Control: no-cache, private
< Vary: Accept-Encoding
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< Transfer-Encoding: chunked
< Content-Type: text/html; charset=UTF-8
< 
<
```

I am getting `200 OK` response from curl and obviously I can access the same link from my browser but I can't actually ping that domain. For example, `ping dev.mysql.com` comes back with 0 success. I think the server is configured to not respond to pings because I tried this external ping utility site, https://www.ipaddressguide.com/ping, and that's also coming back with 100% failures.

Not sure if this is a bug within the linter or if there's room for improvements. Right now my only option is to add that domain to the `ignore` option of this rule.

Running into similar `ECONRESET` error for some other external link too.

```text
34:28   error  https://jcp.org/en/jsr/detail?id=330 is dead. (request to https://jcp.org/en/jsr/detail?id=330 failed, reason: read ECONNRESET)                      no-dead-link
  83:198  error  https://www.osgi.org/developer/specifications/ is dead. (request to https://www.osgi.org/developer/specifications/ failed, reason: read ECONNRESET)  no-dead-link
  26:42  error  https://jax-rs-spec.java.net/ is dead. (request to https://jax-rs-spec.java.net/ failed, reason: read ECONNRESET)  no-dead-link
```

Sometimes the above errors only appear in my local machine and not in the CI pipeline. Sometimes they appear in both.

I just realized the same link gets reported as invalid with different reason. For example:

```text
  34:28  error  https://jcp.org/en/jsr/detail?id=330 is dead. (request to https://jcp.org/en/jsr/detail?id=330 failed, reason: connect ECONNRESET 137.254.60.38:443)  no-dead-link
```

This shows it's failing with `reason: connect ECONNRESET` and the exact same link mentioned in previous comment had `reason: read ECONNRESET`

I'm also getting `ECONREFUSED` error from time to time:

```text
   7:22   error  http://tools.ietf.org/html/rfc6749 is dead. (request to http://tools.ietf.org/html/rfc6749 failed, reason: connect ECONNREFUSED 64.170.98.42:80)                  no-dead-link
  16:140  error  http://tools.ietf.org/html/rfc6749#page-10 is dead. (request to http://tools.ietf.org/html/rfc6749#page-10 failed, reason: connect ECONNREFUSED 64.170.98.42:80)  no-dead-link
```

I just got the chance to try it out. It's probably working for the `ECONNRESET` related errors. The following error disappeared after I updated to `4.4.2`

```text
  83:198  error  https://www.osgi.org/developer/specifications/ is dead. (request to https://www.osgi.org/developer/specifications/ failed, reason: socket hang up)  no-dead-link
```

But, I'm still getting `ECONREFUSED` error from these two links:

```text
   7:22   error  https://tools.ietf.org/html/rfc6749 is dead. (request to https://tools.ietf.org/html/rfc6749 failed, reason: connect ECONNREFUSED 64.170.98.42:443)                  no-dead-link
  16:140  error  https://tools.ietf.org/html/rfc6749#page-10 is dead. (request to https://tools.ietf.org/html/rfc6749#page-10 failed, reason: connect ECONNREFUSED 64.170.98.42:443)  no-dead-link
```

--- a/package.json
+++ b/package.json
@@ -30,8 +30,8 @@
   "dependencies": {
     "fs-extra": "^8.0.1",
     "get-url-origin": "^1.0.1",
-    "isomorphic-fetch": "^2.2.1",
     "minimatch": "^3.0.4",
+    "node-fetch": "^2.6.0",
     "textlint-rule-helper": "^2.1.1"
   },
   "lint-staged": {

--- a/src/no-dead-link.js
+++ b/src/no-dead-link.js
@@ -72,6 +72,7 @@ function isIgnored(uri, ignore = []) {
  * @return {{ ok: boolean, redirect?: string, message: string }}
  */
 async function isAliveURI(uri, method = 'HEAD') {
+  const { host } = URL.parse(uri);
   const opts = {
     method,
     // Disable gzip compression in Node.js
@@ -83,13 +84,15 @@ async function isAliveURI(uri, method = 'HEAD') {
     // https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/111
     headers: {
       'User-Agent': 'textlint-rule-no-dead-link/1.0',
-      'Accept': '*/*'
+      'Accept': '*/*',
+      // Same host for target url
+      // https://github.com/textlint-rule/textlint-rule-no-dead-link/issues/111
+      'Host': host
     },
     // Use `manual` redirect behaviour to get HTTP redirect status code
     // and see what kind of redirect is occurring
     redirect: 'manual',
   };
-
   try {
     const res = await fetch(uri, opts);
 
@@ -253,7 +256,7 @@ function reporter(context, options = {}) {
       if (typeof node.url === 'undefined') {
         return;
       }
-      
+
       // [text](http://example.com)
       //       ^
       const index = node.raw.indexOf(node.url) || 0;

--- a/src/no-dead-link.js
+++ b/src/no-dead-link.js
@@ -1,5 +1,5 @@
 import { RuleHelper } from 'textlint-rule-helper';
-import fetch from 'isomorphic-fetch';
+import fetch from 'node-fetch';
 import URL from 'url';
 import fs from 'fs-extra';
 import minimatch from 'minimatch';

--- a/test/no-dead-link.js
+++ b/test/no-dead-link.js
@@ -16,7 +16,7 @@ tester.run('no-dead-link', rule, {
     'should success with retrying on error: [npm results for textlint](https://www.npmjs.com/search?q=textlint)',
     'should treat 200 OK as alive: https://httpstat.us/200',
     'should treat 200 OK. It require User-Agent: Navigate to [MySQL distribution](https://dev.mysql.com/downloads/mysql/) to install MySQL `5.7`.',
-    'should treat 200 OK. It require User-Agent: <https://tools.ietf.org/html/rfc6749>',
+    'should treat 200 OK. It require User-Agent: https://tools.ietf.org/html/rfc6749',
     {
       text:
         'should be able to check a URL in a plain text: https://example.com/',

